js/wp-autoupdates.js

```diff
@@ -1,286 +1,88 @@
-			} else {
-				var errorHTML = add_error_notice( html, response.data.error );
-				wp.a11y.speak( response.data.error, 'polite' );
-				$parent.html( errorHTML );
-			}
-			
-		})
-		.fail(function (response) {
-			var errorHTML = add_error_notice( html, wp_autoupdates.auto_update_error );
-			wp.a11y.speak( wp_autoupdates.auto_update_error, 'polite' );
-			$parent.html( errorHTML );
-		})
-		.always(function (response) {
-		});
-	});
-});
\ No newline at end of file
+		label.text( 'enable' === action ? wp_autoupdates.enabling : wp_autoupdates.disabling );
+		anchor.find( '.dashicons-update' ).removeClass( 'hidden' );
+
+		data = {
+			action: 'toggle_auto_updates',
+			_ajax_nonce: href._wpnonce,
+			state:  action,
+			type: type,
+			asset: 'plugin' === type ? href.plugin : href.theme,
+		};
+
+		$.post( ajaxurl, data )
+			.done( function( response ) {
+				if ( response.success ) {
+					// Update the counts in the enabled/disabled views if on on
+					// screen with a list table.
+					// TODO: If either count started out 0 the appropriate span won't
+					//       be there and hence won't be updated.
+					if ( 'themes' !== pagenow ) {
+						var enabled        = $( '.autoupdate_enabled span' ),
+							disabled       = $( '.autoupdate_disabled span' ),
+							enabledNumber  = parseInt( enabled.text().replace( /[^\d]+/g, '' ) ) || 0,
+							disabledNumber = parseInt( disabled.text().replace( /[^\d]+/g, '' ) ) || 0;
+
+						switch ( action ) {
+							case 'enable':
+								++enabledNumber;
+								--disabledNumber;
+
+								break;
+							case 'disable':
+								--enabledNumber;
+								++disabledNumber;
+								break;
+						}
+
+						enabledNumber  = Math.max( 0, enabledNumber );
+						disabledNumber = Math.max( 0, disabledNumber );
+
+						enabled.text( '(' + enabledNumber + ')' );
+						disabled.text( '(' + disabledNumber + ')' );
+					}
+
+					if ( 'enable' === action ) {
+						anchor.removeClass( 'enable' ).addClass( 'disable' );
+						label.text( wp_autoupdates.disable );
+						parent.find( '.auto-update-time').removeClass( 'hidden' );
+					} else {
+						anchor.removeClass( 'disable' ).addClass( 'enable' );
+						label.text( wp_autoupdates.enable );
+						parent.find( '.auto-update-time').addClass( 'hidden' );
+					}
+
+					wp.a11y.speak( 'enable' === action ? wp_autoupdates.enabled : wp_autoupdates.disabled, 'polite' );
+				} else {
+					parent.find( '.auto-updates-error' ).removeClass( 'hidden' ).addClass( 'notice error' ).find( 'p' ).text( response.data.error );
+					wp.a11y.speak( response.data.error, 'polite' );
+				}
+			} )
+			.fail( function( response ) {
+				parent.find( '.auto-updates-error' ).removeClass( 'hidden' ).addClass( 'notice error' ).find( 'p' ).text( wp_autoupdates.auto_update_error );
+				wp.a11y.speak( wp_autoupdates.auto_update_error, 'polite' );
+			} )
+			.always( function() {
+				anchor.find( '.dashicons-update' ).addClass( 'hidden' );
+			} );
+	} );
+} )( jQuery );
```

Comment:
Just a nitpick but when merging to core perhaps better to pass `( window.jQuery, window.pagenow )` and run this on DOM ready.

Reply:
agree on pagenow, will fix that.

on jQuery, most of the core JS I've looked at just uses `jQuery`, e.g., `wp-admin/js/updates.js`)